### PR TITLE
Feature/display activity cards

### PR DESCRIPTION
**Pull request by Asejs**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "name": "react-flask-app",
   "version": "0.1.0",
   "private": true,
-  "proxy": "http://localhost:5000",
+  "proxy": "http://127.0.0.1:5000",
```
Comment (CsabaTheKiss) [inline package.json:5]: Why was this changed? 🤔 Most likely I don't see the full picture. I would think that we want to run our things on localhost when developing.

File: public/index.html [hunk 2]
```diff
@@ -9,7 +9,6 @@
       name="description"
       content="Web site created using create-react-app"
     />
-    <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
```
Comment (CsabaTheKiss) [inline public/index.html:12]: So this isn't needed anymore? Don't we need a logo here?

File: src/App.css [hunk 3]
```diff
@@ -35,27 +36,41 @@ a:hover {
...
 
 h4 {
   color: #727272;
-  font-size: 1.0em;
+  font-size: 1.3em;
   line-height: 160%;
 }
 
+.logo {
+  width: 190px;
```
Comment (CsabaTheKiss) [inline src/App.css:61]: Just wondering: why is `px` (absolute measurment) used for sizes (width, padding, top, etc.) and `em` (relative measurment) for fonts? 🤔

File: src/App.tsx [hunk 4]
```diff
@@ -1,17 +1,17 @@
...
 import './App.css';
 
 function App() {
   return (
       <div className="App">
         <Layout>
 				  <Routes>
-            <Route path='/' element={<Home />} />
-            <Route path='/aktiviteter' element={<Activities />} />
+            <Route path='/' element={<HomePage />} />
```
Comment (CsabaTheKiss) [inline src/App.tsx:13]: I like the name change here! 👍 (adding `Page` to the component names)

File: src/api/activities.api.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { baseURL } from "../config";
+import { ActivityItem } from "./activities.api.types";
+
+export async function getAllActivities(): Promise<{data: ActivityItem[]; error: any}> {
+    try {
+      const response = await fetch(baseURL + "activities_get")
```
Comment (CsabaTheKiss) [inline src/api/activities.api.ts:6]: Maybe the string concatenation can be bit more readable (getting rid of the + operatir) with "template string":
`baseURL + "activities_get"` => `` `${baseURL} activities_get` ``

File: src/components/Activity.tsx [hunk 6]
```diff
@@ -1,52 +1,20 @@
...
-      setActivities(json)
-      console.log("activities", json)
-    } catch (error) {
-      setError(error);
-      console.log("error")
-    }
-    setLoading(false);
-  }
+    console.warn('allActivities', allActivities)
+  }, [allActivities])
```
Comment (CsabaTheKiss) [inline src/components/Activity.tsx:11]: Is this useEffect still needed?

File: src/components/ActivityCard.tsx [hunk 7]
```diff
@@ -1,45 +1,29 @@
...
 
-  const handleExpandClick = () => {
-    setExpanded(!expanded);
-  };
+export default function ActivityCard(props: ActivityCardProps) {
+  const { activityItem } = props;
 
   return (
-    <Card sx={{ mb: '30px'}}>
+    <Card sx={{ width: '900px', ml: 'auto', mr: 'auto', mb: '30px'}}>
```
Comment (CsabaTheKiss) [inline src/components/ActivityCard.tsx:26]: It was quite some time when I last used react mui: is it possible to extract these styles somehow? 🤔

File: src/components/ActivityCard.tsx [hunk 8]
```diff
@@ -50,52 +34,37 @@ export default function ActivityCard() {
...
-        subheader="Date"
+        title={activityItem.username}
       />
       <CardMedia
         component="img"
-        height="290"
-        image={ defaultimg }
-        alt="activityalt"
+        height="390"
+        src={"assets/" + activityItem.image }
```
Comment (CsabaTheKiss) [inline src/components/ActivityCard.tsx:42]: See my prev. comment: https://github.com/Asejs/Activity-Planner-React/pull/2/files#r954550959
Would suggest to use string template instead of classic string concatenation.

File: src/components/AddActivity.tsx [hunk 9]
```diff
@@ -26,85 +28,90 @@ export default function AddActivity() {
...
+                          label="Tittel på aktivitet"
+                        />
+
+                        <TextField
+                          id="outlined-number"
+                          label="Dato"
+                          type="date"
+                          InputLabelProps={{
+                              shrink: true,
+                        }}
```
Comment (CsabaTheKiss) [inline src/components/AddActivity.tsx:55]: Minor comment: as no prettyfier is applied in your VS code, these closing brackets are misaligned.
They should be aligned to `InputLabelProps`, while now they are aligned to `<TextField` (making it harder to read => harder to see where a block starts and ends)

File: src/components/AddActivity.tsx [hunk 10]
```diff
@@ -26,85 +28,90 @@ export default function AddActivity() {
...
                 </Grid>
-
                 <Grid item xs={2}>
                   <input type="file"
-                  id="fileUploadButton"
-                  style={{ display: 'none' }}
-                  // onChange={onFileChange}
+                    id="fileUploadButton"
+                    style={{ display: 'none' }}
+                    // onChange={onFileChange}
```
Comment (CsabaTheKiss) [inline src/components/AddActivity.tsx:84]: Is this commented line still needed?

File: src/components/AddActivity.tsx [hunk 11]
```diff
@@ -26,85 +28,90 @@ export default function AddActivity() {
...
-
                 <Grid item xs={2}>
-                    Bilde alternativer
+                    <p>Bilde alternativer</p>
                 </Grid>
               </Grid>
 
-              <Box  sx={{ display: 'flex', justifyContent: 'flex-end'}}>
-                <Button variant='contained' sx={{ backgroundColor: 'darkgrey' }}>Legg til aktivitet</Button>
+              <Box sx={{ display: 'flex', justifyContent: 'flex-end'}}>
```
Comment (CsabaTheKiss) [inline src/components/AddActivity.tsx:108]: As I mentioned: it was quite a while since I use react mui. This is the only way to style these mui components?
(preferably these should be separated in their own css / scss etc. file)

File: src/components/AppBar.tsx [hunk 12]
```diff
@@ -124,6 +115,7 @@ const ResponsiveAppBar: React.FC = () => {
                 </Tooltip>
                 
                 <Menu
+                disableScrollLock={true}
```
Comment (CsabaTheKiss) [inline src/components/AppBar.tsx:118]: This could be just `disableScrollLock`, in theory you don't need to add `true`, as value.
The mere presence of the attribute should be enough to say, that it's value is `true`.

File: src/pages/homePage/HomePage.css [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+.home_description {
+    margin-right: auto;
+    margin-left: auto;
+    position: relative;
+    top: 400px;
+    color: white;
+    background-color: #789778;
+    border-radius: 10px;
+    padding: 100px;
+    box-shadow: rgba(0, 0, 0, 0.2) 0px 0px 0px 1px;
```
Comment (CsabaTheKiss) [inline src/pages/homePage/HomePage.css:25]: If I remember correctly, when you write`0` you don't need to add the unit `px` afterwards.

File: src/pages/homePage/HomePage.tsx [hunk 14]
```diff
@@ -0,0 +1,18 @@
...
+
+const HomePage: React.FC = () => {
+  return (
+    <>
+      <div className="homepage" style={{ overflow: 'hidden' }} />
+
+      <div className="home_description">
+          Velkommen til<br />
+          <img src="assets/logo.png" alt="logo"/><br />
+          Logg inn for å melde deg på aktiviteter!
```
Comment (CsabaTheKiss) [inline src/pages/homePage/HomePage.tsx:12]: Would suggest to not use <br> elements to break lines.
(not so good semantic meaning for those, and harder to style. Also it's better to group your things in logical blocks)

I would suggest to use <p> instead.

Comment (Asejs) [reply src/pages/homePage/HomePage.tsx:12]: Changed to <p> instead.

File: src/pages/homePage/HomePage.css (hunk 13 above)
Comment (Asejs) [reply src/pages/homePage/HomePage.css:25]: True, I removed the unit `px`.

File: src/components/AppBar.tsx (hunk 12 above)
Comment (Asejs) [reply src/components/AppBar.tsx:118]: Changed.

File: package.json (hunk 1 above)
Comment (Asejs) [reply package.json:5]: True, I changed it back to localhost.

File: src/api/activities.api.ts (hunk 5 above)
Comment (Asejs) [reply src/api/activities.api.ts:6]: Changed.

File: src/components/Activity.tsx (hunk 6 above)
Comment (Asejs) [reply src/components/Activity.tsx:11]: No, it's not. Removed it.

File: src/components/AddActivity.tsx (hunk 9 above)
Comment (Asejs) [reply src/components/AddActivity.tsx:55]: Can you show me how to apply the prettyfier in VS code?

File: src/App.css (hunk 3 above)
Comment (Asejs) [reply src/App.css:61]: I have been a bit inconsistent there 😅  What should I use?

File: package.json (hunk 1 above)
Comment (CsabaTheKiss) [reply package.json:5]: Interesting 🤔 Maybe this is needed by the local flask BE?
But no worries, I was just wondering about this change.

File: src/components/AddActivity.tsx (hunk 9 above)
Comment (CsabaTheKiss) [reply src/components/AddActivity.tsx:55]: Sure! I'll check what setup I have and I'll show you.

File: src/App.css (hunk 3 above)
Comment (CsabaTheKiss) [reply src/App.css:61]: I would say: it depends. Every team / project has it's own rules. Consistency in general is just a good rule to follow.
Maybe on a project the team agrees that they would use `em` for fonts (for a given reason).

So in this case it's up to you (as it's your project).

As I wrote: I was just wondering if there's some reasoning behind this.

**Comment by CsabaTheKiss** [comment]
As discussed: the css extraction / changes are possible, but we don't want to focus on this now.

**Review by CsabaTheKiss — APPROVED** [review]
Bra jobbe! 👏
